test/src/test/java/aima/test/unit/search/adversarial/SearchForAdversarialActionFunctionTest.java

```diff
@@ -28,22 +29,23 @@
 public class SearchForAdversarialActionFunctionTest {
 	@Parameters(name = "{index}: {0}")
 	public static Collection<Object[]> implementations() {
-		return Arrays.asList(new Object[][] { { "MinimaxDecision" },  { "AlphaBetaSearch" } });
+		return Arrays.asList(new Object[][]{{"MinimaxDecision"}, {"AlphaBetaSearch"}, {"IterativeDeepeningAlphaBetaSearch"}});
 	}
 
 	@Parameter
 	public String searchFunctionName;
 
-	public <S, A, P> A searchForAdversarialAction(Game<S, A, P> game) {
-		SearchForAdversarialActionFunction<S, A> searchFn = null;
-		
-		if ("MinimaxDecision".equals(searchFunctionName)) {
-			searchFn = new MinimaxDecision<>(game);
+	public <S, A, P> A searchForAdversarialAction(Game<S, A, P> game, Integer... params) {
+		SearchForAdversarialActionFunction<S, A> searchFn;
+		if (params != null) {
+			if ("MinimaxDecision".equals(searchFunctionName)) {
+				searchFn = new MinimaxDecision<>(game);
+			} else {
+				searchFn = new AlphaBetaSearch<>(game);
+			}
+		} else {
+			searchFn = new IterativeDeepeningAlphaBetaSearch<>(game, params[0], params[1], params[2]);

```

Comment:
NOTE: IterativeDeepeningAlphaBetaSearch is never tested! At this point params is null and the way the test below is configured AlphaBetaSearch is all that is called and only one test contains the params for the IterativeDeepening calls. This code should be re-configured to just be:

	public <S, A, P> A searchForAdversarialAction(Game<S, A, P> game, Integer... params) {
		SearchForAdversarialActionFunction<S, A> searchFn;
		
		if ("MinimaxDecision".equals(searchFunctionName)) {
			searchFn = new MinimaxDecision<>(game);
		} else if ("AlphaBetaSearch".equals(searchFunctionName)) {
			searchFn = new AlphaBetaSearch<>(game);
		} else {
			searchFn = new IterativeDeepeningAlphaBetaSearch<>(game, params[0], params[1], params[2]);
		}
		
		return searchFn.apply(game.initialState());
	}